packages/compass/src/main/squirrel-startup.ts

```diff
@@ -0,0 +1,80 @@
+import path from 'path';
+import { spawn } from 'child_process';
+import { createLogger } from '@mongodb-js/compass-logging';
+const { debug } = createLogger('SQUIRREL-STARTUP');
+
+function runUpdateExe(args: string[]): Promise<void> {
+  return new Promise<void>(function (resolve) {
+    const updateExe = path.resolve(
+      path.dirname(process.execPath),
+      '..',
+      'Update.exe'
+    );
+    debug('Spawning `%s` with args `%s`', updateExe, args.join(' '));
+    spawn(updateExe, args, {
+      detached: true,
+    }).on('close', () => resolve());
+  });
+}
+
+function getShortcutName() {
+  // The shortcut name is the app name plus the channel name if it is not
+  // stable. ie. "MongoDB Compass Readonly Beta"
+
+  const parts: string[] = [process.env.HADRON_PRODUCT_NAME];
+  if (process.env.HADRON_CHANNEL !== 'stable') {
+    parts.push(
+      process.env.HADRON_CHANNEL.charAt(0).toUpperCase() +
+        process.env.HADRON_CHANNEL.slice(1)
+    );
+  }
+  return parts.join(' ');
+}
+
+/*
+Squirrel will spawn Compass with command line flags on first run, updates, and
+uninstalls. It is very important that we handle these events as early as
+possible, and quit immediately after handling them. Squirrel gives apps a short
+amount of time (~15sec) to apply these operations and quit.
+*/
+export async function handleSquirrelWindowsStartup(): Promise<boolean> {
+  if (process.platform !== 'win32') {
+    return false;
+  }
+
+  const shortcutName = getShortcutName();
+
+  const cmd = process.argv[1];
+  debug('processing squirrel command `%s`', cmd);
+
+  /*
+  For more detailed info on these commands, so Electron and Squirrel Windows'
+  documentation:
```

Comment:
It is a pity that electron-userland/electron-installer-squirrel-windows [links back to electron-squirrel-startup](https://github.com/electron-userland/electron-installer-squirrel-windows?tab=readme-ov-file#integration).

I think that README would be a lot clearer if they just put this code there as an example, linked to Squirrel's documentation and maybe explained what these three or four CLI args do.

Reply:
IIRC the https://github.com/mongodb-js/electron-squirrel-startup is used by some other folks. If we're no longer using it we might want to mark it unmaintained/archived or something. Could be worth bringing up in the compass channel.
FWIW I'm all for removing small packages and also having more typescript!